### ISSUE TITLE
C# Dev Kit hot reload does not work

### ISSUE THREAD

**Issue by mlin01** [microsoft/vscode-dotnettools#1702]
Type: <b>Bug</b>

The hot reload does not work in my projects.

Extension version: 1.14.14
VS Code version: Code 1.96.0 (138f619c86f1199955d53b4166bef66ef252935c, 2024-12-11T02:29:09.626Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i9-12900H (20 x 2918)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|63.68GB (40.90GB free)|
|Process Argv|--crash-reporter-id a120f852-a8c3-41c2-9f90-a3345e4da3c5|
|Screen Reader|no|
|VM|29%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
vscaat:30438848
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupytercf:31046870
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc2:31192216
cf971741:31144450
iacca1:31171482
notype1:31157159
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624

```

</details>

**Comment by klukc7** [comment]
For me it is also not working...

**Comment by spadapet** [comment]
@mlin01 can you give details on your project and exactly what isn't working with hot reload? It's best if we have steps to try and reproduce the same error. Thanks!

**Comment by NoahStahl** [comment]
@spadapet One example:

1. Follow the getting started tutorial for a blazor app, exactly as written: https://dotnet.microsoft.com/en-us/learn/aspnet/blazor-tutorial/install
2. Once you've completed the `Run your app` step, confirm your browser launches with the home content of "Hello, world!"
3. In the `Home.razor` component file, change the text "Hello, world" to something different. Save your changes.
4. Observe the browser content has not changed to reflect the changed text.
5. In VSCode output, see this message indicating code changes we recognized:
```
2025-03-29 17:29:29.719 [info]  💡 Possible code changes were found.
2025-03-29 17:29:33.103 [info]  Code changes were successfully applied.
```
6. If you manually refresh the browser, the changed text is rendered.

The point of hot reload is expected to be a rerender without manual F5, correct? The page should refresh itself on save.

**Comment by ryzngard** [comment]
> The point of hot reload is expected to be a rerender without manual F5, correct? The page should refresh itself on save.

Correct. We have an issue where the js needed to refresh automatically isn't hooked up like in VS. It's one of the blocking items before we can make hot reload not "experimental" in VS Code.

**Comment by anthonymastreanvae** [comment]
So, the update here from Feb 11 should not be interpreted as hot reload works for Razor/Blazor?

https://devblogs.microsoft.com/dotnet/csharp-on-visual-studio-code-just-got-better-with-enhancements-to-csharp-dev-kit/#razor/blazor-experience
